bokeh/properties.py

```diff
@@ -837,7 +837,8 @@ class List(Seq):
 
     """
 
-    def __init__(self, item_type, default=[], help=None):
+    def __init__(self, item_type, default=None, help=None):
+        default = default or []
```

Comment:
@mattpap I only have a vague recollection of what is going on here, a slightly more detailed explanation would be appreciated.